eclair-core/src/main/scala/fr/acinq/eclair/blockchain/fee/BitcoinCoreFeeProvider.scala

```diff
@@ -47,30 +47,31 @@ class BitcoinCoreFeeProvider(rpcClient: BitcoinJsonRPCClient, defaultFeerates: F
     blocks_36 <- estimateSmartFee(36)
     blocks_72 <- estimateSmartFee(72)
     blocks_144 <- estimateSmartFee(144)
+    blocks_1008 <- estimateSmartFee(1008)
   } yield FeeratesPerKB(
-    block_1 = if (block_1 > 0) block_1 else defaultFeerates.block_1,
-    blocks_2 = if (blocks_2 > 0) blocks_2 else defaultFeerates.blocks_2,
-    blocks_6 = if (blocks_6 > 0) blocks_6 else defaultFeerates.blocks_6,
-    blocks_12 = if (blocks_12 > 0) blocks_12 else defaultFeerates.blocks_12,
-    blocks_36 = if (blocks_36 > 0) blocks_36 else defaultFeerates.blocks_36,
-    blocks_72 = if (blocks_72 > 0) blocks_72 else defaultFeerates.blocks_72,
-    blocks_144 = if (blocks_144 > 0) blocks_144 else defaultFeerates.blocks_144)
+    block_1 = if (block_1.isValid) block_1 else defaultFeerates.block_1,
+    blocks_2 = if (blocks_2.isValid) blocks_2 else defaultFeerates.blocks_2,
+    blocks_6 = if (blocks_6.isValid) blocks_6 else defaultFeerates.blocks_6,
+    blocks_12 = if (blocks_12.isValid) blocks_12 else defaultFeerates.blocks_12,
+    blocks_36 = if (blocks_36.isValid) blocks_36 else defaultFeerates.blocks_36,
+    blocks_72 = if (blocks_72.isValid) blocks_72 else defaultFeerates.blocks_72,
+    blocks_144 = if (blocks_144.isValid) blocks_144 else defaultFeerates.blocks_144,
+    blocks_1008 = if (blocks_1008.isValid) blocks_1008 else defaultFeerates.blocks_1008)
 }
 
 object BitcoinCoreFeeProvider {
-  def parseFeeEstimate(json: JValue): Long = {
+  def parseFeeEstimate(json: JValue): FeeratePerKB = {
     json \ "errors" match {
       case JNothing =>
         json \ "feerate" match {
           case JDecimal(feerate) =>
             // estimatesmartfee returns a fee rate in Btc/KB
-            btc2satoshi(Btc(feerate)).toLong
+            FeeratePerKB(btc2satoshi(Btc(feerate)))
```

Comment:
The conversion could even be implicit (`FeeratePerKB(Btc(feerate))` works)

```suggestion
            FeeratePerKB(FeeratePerKB(Btc(feerate).toSatoshi))
```

Reply:
True, conversion aren't lossy when going to sats, but I like it explicit!